buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java

```diff
@@ -54,6 +61,40 @@ public void apply(Project project) {
                 }
               });
     }
+    if (System.getenv().containsKey("METALAVA_BINARY_PATH")) {
+      String METALAVA_BINARY_PATH = System.getenv("METALAVA_BINARY_PATH");
+      AndroidSourceSet mainSourceSet = android.getSourceSets().getByName("main");
+      List<String> sourcePaths = new ArrayList<>();
+      for (File directory : mainSourceSet.getJava().getSrcDirs()) {
+        sourcePaths.add(directory.getAbsolutePath());
+      }
+      String sourcePathArgument = String.join(",", sourcePaths);
+      project.getTasks().create("apiInformation", ApiInformationTask.class, task -> {
+        task.setProperty("apiTxt", project.getProjectDir() + "/api.txt");
+        task.setProperty("metalavaBinaryPath", METALAVA_BINARY_PATH);
+        task.setProperty("sourcePath", sourcePathArgument);
+        task.setProperty("outputPath", project.getRootProject().getBuildDir() + "/apiinfo/subproject" + project.getPath());
```

Comment:
can you name files something like "project_path_name_api.txt" instead of "subproject:project:path:name"? It's not clear to me why you need the "subproject" prefix, and colons in paths need a lot of escaping